packages/calypso-e2e/src/lib/pages/reader-page.ts

```diff
@@ -0,0 +1,79 @@
+import { Page } from 'playwright';
+
+const selectors = {
+	readerVisitLink: '.reader-visit-link',
+	shareButtonLocator: '.reader-share__button',
+	firstComboCardPostLocator: '.reader-combined-card__post-title-link',
+	siteContentLocator: '.site-selector__sites .site__content',
+	commentButtonLocator: '.comment-button',
+	commentTextAreaLocator: '.comments__form textarea',
+	commentSubmitLocator: '.comments__form button',
+	commentContentLocator: '.comments__comment-content',
+};
+
+/**
+ * Page representing the Reader Page.
+ */
+export class ReaderPage {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Get the URL of the latest post in Reader
+	 *
+	 * @returns {Promise<string>} String of URL for latest post.
+	 */
+	async siteOfLatestPost(): Promise< string > {
+		const href = await this.page.getAttribute( selectors.readerVisitLink, 'href' );
+		return new URL( href ? href : '' ).host;
+	}
+
+	/**
+	 * Share latest post in Reader
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async shareLatestPost(): Promise< void > {
+		const hasSharablePost = await this.page.$( selectors.shareButtonLocator );
+		if ( ! hasSharablePost ) {
+			// no shareable posts on this screen. try moving into a combined card
+			await this.page.click( selectors.firstComboCardPostLocator );
+		}
```

Comment:
This block from the older code seems a little funky to me - do you think we need this in the newer implementation? Do you think there's a way we can work the test design to always ensure there is a test to share instead of doing this conditional logic?